### PR TITLE
Copyright header check

### PR DESCRIPTION
**Pull request by sleeepyjack**
Alternative to #254, which uses a `pre-commit` hook.

### CODE REVIEW

**Comment by PointKernel** [comment]
Is this the ChatGPT-generated solution?

File: ci/checks/copyright.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
```
Comment (PointKernel) [inline ci/checks/copyright.py:1]: The copyright file itself doesn't have a copyright...

🥲

Comment (sleeepyjack) [reply ci/checks/copyright.py:1]: Lol, my bad

**Comment by sleeepyjack** [comment]
> Is this the ChatGPT-generated solution?

I can neither confirm nor deny this ;) Let's say it's based on what ChatGPT suggested.

The only thing left that I need to figure out is how to pipe the paths of those files that have been changed to the script via pre-commit.

**Comment by sleeepyjack** [comment]
Hmm, the hook works as intended when run locally, i.e., `git commit` triggers the `pre-commit` hook, but CI still complains:
```
copyright-check..........................................................Failed
- hook id: copyright-check
- duration: 0.03s
- exit code: 1

warning: Copyright line in include/cuco/static_multimap.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/static_map_kernels.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/hash_functions.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/static_map.inl is not up-to-date (2022 != 2023)
warning: Copyright line in examples/static_map/custom_type_example.cu is not up-to-date (2022 != 2023)
warning: Copyright line in cmake/thirdparty/get_thrust.cmake is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/prime.hpp is not up-to-date (2022 != 2023)
warning: Copyright line in ci/checks/doxygen.sh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/bitwise_compare.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/utils.cuh is not up-to-date (2022 != 2023)
warning: Copyright line is missing from benchmarks/analysis/notebooks/Utils.py
warning: Copyright line in include/cuco/static_map.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/pair.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/static_multimap/kernels.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in cmake/thirdparty/get_libcudacxx.cmake is not up-to-date (2021 != 2023)
warning: Copyright line in benchmarks/hash_table/dynamic_map_bench.cu is not up-to-date (2022 != 2023)
warning: Copyright line in examples/static_map/device_view_example.cu is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/allocator.hpp is not up-to-date (2022 != 2023)
warning: Copyright line in examples/static_map/host_bulk_example.cu is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/dynamic_map.inl is not up-to-date (2022 != 2023)
warning: Copyright line in ci/gpu/build.sh is not up-to-date (2021 != 2023)
warning: Copyright line in include/cuco/traits.hpp is not up-to-date (2021 != 2023)
warning: Copyright line in tests/utils.cuh is not up-to-date (2021 != 2023)
warning: Copyright line in include/cuco/detail/static_multimap/device_view_impl.inl is not up-to-date (2022 != 2023)
warning: Copyright line in ci/checks/style.sh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/probe_sequence_impl.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/probe_sequences.cuh is not up-to-date (2021 != 2023)
warning: Copyright line in include/cuco/sentinel.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/hash_functions.cuh is not up-to-date (2022 != 2023)
warning: Copyright line in include/cuco/detail/dynamic_map_kernels.cuh is not up-to-date (2022 != 2023)

>>>> FAILED: copyright check
```
It seems CI is passing in all files instead of only the files that have been changed. Where do we trigger this hook in CI? I can't seem to find it.

**Pull request closed without merging**